client/app/pages/queries/QuerySource.jsx

```diff
@@ -244,13 +251,13 @@ function QuerySource(props) {
       schedule: query.schedule,
       refreshOptions,
     }).result.then(schedule => {
-      updateQuery(query, { schedule }).then(setQuery);
+      updateQuerySchedule(query, schedule).then(setQuery);
```

Comment:
I was thinking about putting the whole Dialog call in an action, but NP :slightly_smiling_face:

Reply:
@gabrieldutra I changed implementation of QueryMetadata component so now it can display data source too (https://github.com/getredash/redash/pull/4476/commits/32bd46e88d9a24770563bf9e2d82d6c8783ce6de). I temporarily added data source to Query Source page so you can try in on preview instance of this PR (I'll remove it before merging :slightly_smiling_face:). It slightly differs from all existing variants, but not too much and IMO works good enough (check how it stacks on different screen widths)

Reply:
Let it stay here for now. Probably I'll wrap all dialogs into some hook; maybe I'll even group state/callbacks/dialogs in few hooks to make this component less crowded - will see later